hackathon_site/hardware/serializers.py

```diff
@@ -204,6 +208,16 @@ def create(self, validated_data):
         )
         new_order = None
         response_data = {"order_id": None, "hardware": [], "errors": []}
+
+        # The reason why doing this is because the id field stores the hardware object, django cannot translate hardware object into JSON. Therefore, loop has been used to call Profile seriliazer and assign the JSON file to the validated_data filed.
+        counter = 0
+        for item in validated_data["hardware"]:
```

Comment:
This block is unnecessary since you can just use `self.data`, but for future reference a cleaner way to write it would have been:

```python
for i, item in enumerate(validated_data["hardware"]):
    validated_data["hardware"][i] = HardwareSerializer(item["id"]).data
```